wallets/metamask/src/fixture-actions/noCacheMetaMaskSetup.ts

```diff
@@ -0,0 +1,89 @@
+import path from 'node:path'
+import { type BrowserContext, chromium } from '@playwright/test'
+import appRoot from 'app-root-path'
+import axios from 'axios'
+import fs from 'fs-extra'
+import unzipper from 'unzipper'
+import { DEFAULT_METAMASK_VERSION, EXTENSION_DOWNLOAD_URL } from '../utils/constants'
+
+async function prepareMetaMask(version: string = DEFAULT_METAMASK_VERSION): Promise<string> {
+  const downloadsDirectory =
+    process.platform === 'win32' ? appRoot.resolve('/node_modules') : path.join(process.cwd(), 'downloads')
+  await fs.ensureDir(downloadsDirectory)
+
+  const metamaskDirectory = path.join(downloadsDirectory, `metamask-chrome-${version}.zip`)
+  const archiveFileExtension = path.extname(metamaskDirectory)
+  const outputPath = metamaskDirectory.replace(archiveFileExtension, '')
+  const metamaskManifestPath = path.join(outputPath, 'manifest.json')
+
+  if (!fs.existsSync(metamaskManifestPath)) {
+    await downloadAndExtract(EXTENSION_DOWNLOAD_URL, metamaskDirectory)
+  }
+
+  return outputPath
+}
+
+async function downloadAndExtract(url: string, destination: string): Promise<void> {
+  const response = await axios.get(url, { responseType: 'stream' })
+  const writer = fs.createWriteStream(destination)
+  response.data.pipe(writer)
+  await new Promise((resolve) => writer.on('finish', resolve))
+
+  await unzipArchive(destination)
+}
+
+async function unzipArchive(archivePath: string): Promise<void> {
```

Comment:
This function seems to be unnecessary. I think we can use the archive directly.

Reply:
Couldn't use the archive during testing, reason I need to unzip is to get the manifest.json

Reply:
Dig around the code a bit and you'll see